lib/artifact.js

```diff
@@ -112,25 +64,17 @@ module.exports = (bluebird, fs, logger, request, requestOptions, requestPromise,
          * @return {Promise.<Object>}
          */
         download() {
-            var options;
-
             logger.debug(`In ${this.constructor.name}.download()`);
-            options = requestOptions.createOptions(this.authToken);
-            options.json = true;
-            options.url = this.uri;
-            logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
-
-            return requestPromise.get(options).then((response) => {
-                logger.debug(`In ${this.constructor.name}.download() -> request.get().then()`);
-                logger.debug(`Response: ${JSON.stringify(response, null, 4)}`);
-                logger.debug(`Status code: ${response.statusCode}`);
-
-                return response.body;
-            }, (reason) => {
-                logger.debug(`In ${this.constructor.name}.download() -> request.get().catch()`);
-                logger.debug(`Reason: ${JSON.stringify(reason, null, 4)}`);
-
-                responseHandler.handleErrorResponse(reason);
+
+            return this.shelfRequest.get(this.uri).then((response) => {
+                /*
+                 * In the future we should return a buffer instead of a
+                 * utf8 encoded string. I am only doing this here now
+                 * because the change is unrelated to the feature I
+                 * am working on right now and I don't want to try to
+                 * sneak in something that breaks the interface that bad.
+                 */
+                return response.body.toString("utf8");
```

Comment:
It all depends on how it is stored I suppose. I wouldn't expect it to be that much larger though I guess. I will make a note of it.